demos/link-prediction/random-walks/cora-lp-demo.ipynb

```diff
@@ -4,7 +4,7 @@
    "cell_type": "markdown",
```

Comment:
<p>I think the use of a list here is a good improvement, but I had to re-read it several times to understand all the moving parts, and particularly how the "splits"/"sets" connect to the graphs. Maybe it could be expanded (sorry for the poor formatting, reviewnb's markdown editor kinda sucks):</p><p> </p><p>---</p><p> </p><p>We're going to tackle link prediction as a supervised learning problem on top of node representations/embeddings. The embeddings are computed with the unsupervised node2vec algorithm. After obtaining embeddings, a binary classifier can be used to predict a link, or not, ... [rest of first paragraph]</p><p> </p><p>There's four steps:</p><p> </p><ol><li>Obtain embeddings for each node</li><li>For each set of hyperparameters, train a classifier</li><li>Select the classifier that performs the best</li><li>Evaluate the selected classifier on unseen data to validate its ability to generalise</li></ol><p> </p><p>We have to carefully split the data to avoid data leakage and evaluate the algorithms correctly data:</p><p> </p><ul><li>For computing node embeddings, a <strong>train graph</strong></li><li>For training classifiers, a <strong>classifier training set</strong> of positive and negative edges that weren't used for computing node embeddings</li><li>For choosing the best classifier, an <strong>model selection test set</strong> of positive and negative edges that weren't used for computing node embeddings or training the classifier</li><li>For the final evaluation, a <strong>test graph</strong> with more edges than the train graph to compute test node embeddings, and a <strong>test set</strong> of positive and negative edges not used for computing the test node embeddings or for classifier training or model selection</li></ul><p> </p><p>---</p><p> </p><p>The bolded things should then match the names of the variables (but I think you've already done that?)</p><p> </p><p>Potentially the first paragraph and the "there's four steps" description can move to the introduction too and this section just focuses on the splits.</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/stellargraph/stellargraph/pull/1190/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Reply:
Thanks, yeah I think it makes sense to have the first paragraph + "four steps..." in the introduction, so I've moved it there now - also helps to break up this big markdown section. 

The variable name vs how I refer to the splits in english were slightly different (**Train Graph** vs `graph_train`) so I've put the variable names in parens too.